server/gameObjects/Player.mjs

```diff
@@ -251,12 +261,34 @@ export default class Player {
     );
   }
 
+  hasRangedWeapon() {
+    return this.getWeaponRange() > 1;
```

Comment:
This should be based on weapon type directly, as some melee weapons could have range higher than 1